nodejs_container/native/src/config.rs

```diff
@@ -0,0 +1,137 @@
+use holochain_container_api::{
+    config::{
+        AgentConfiguration, Configuration, DNAConfiguration, InstanceConfiguration,
+        LoggerConfiguration, StorageConfiguration,
+    },
+    Holochain,
+};
+use holochain_net::p2p_config::P2pConfig;
+use neon::{context::Context, prelude::*};
+use std::{collections::HashMap, convert::TryFrom, path::PathBuf};
+
+#[derive(Serialize, Deserialize)]
+pub struct AgentData {
+    pub name: String,
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct DnaData {
+    pub path: PathBuf,
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct InstanceData {
+    pub agent: AgentData,
+    pub dna: DnaData,
+}
```

Comment:
Why are you using these stripped down copies instead of exposing the types in container_api/config?

Reply:
@lucksus the idea was to make it as easy as possible to create a scenario test (e.g. no need to pick and match IDs), but I think you're right and I should just expose the Configuration types and recreate the ConfigBuilder in pure JS. I can also expose the config-from-toml-string functionality to make it fully functional.

Reply:
I just tried exposing the types (`AgentConfiguration`, `DnaConfiguration`, etc) and realized it wouldn't be very helpful, because the exposed `Habitat` constructor expects a JS object anyway, not special opaque rust-backed types. So if the user wants to build a config by hand they have to build an object that matches that shape anyway.

I also made Habitat able to accept a string of TOML config.